### PR TITLE
Make block signal driven

### PR DESCRIPTION
**Pull request by mattdodge**
Also separates `WALK` and `GET` commands

### CODE REVIEW

File: snmp_base_block.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+          Returns the data required by cmdGen for executing SNMP GET
+          operations on an Agent
+        """
+        raise NotImplementedError()
+
+    def make_snmp_request(self, oids):
+        """ Override this in the child block to make the proper request """
+        raise NotImplementedError()
+
+    def _handle_data(self, varBinds, starting_signal):
```
Comment (hansmosh) [inline snmp_base_block.py]: I see that varBinds is to stick with what pysnmp looks like but what do you think about switching that to var_binds?

File: snmp_base_block.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            self._handle_data(result, Signal())
+
+    def _create_data(self):
+        """ Method to be override in inherited classes
+          Returns the data required by cmdGen for executing SNMP GET
+          operations on an Agent
+        """
+        raise NotImplementedError()
+
+    def make_snmp_request(self, oids):
```
Comment (hansmosh) [inline snmp_base_block.py]: Being a little picky here, but why not make make_snmp_request _make_snmp_request?

File: snmp_base_block.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+    agent_host = StringProperty(title="SNMP Agent Url", default='127.0.0.1')
+    agent_port = IntProperty(title="SNMP Agent Port", default=161)
+    timeout = TimeDeltaProperty(title='SNMP GET Timeout',
+                                default={"seconds": 1})
+    retries = IntProperty(title="SNMP GET Retries", default=5)
+    exclude_existing = BoolProperty(
+        title="Exclude Existing Values", default=False)
+    lookup_names = BoolProperty(title="Look up OId names", default=False)
+    lookup_values = BoolProperty(title="Look up OId values", default=False)
```
Comment (hansmosh) [inline snmp_base_block.py]: Shouldn't the 'd' be capital here? OID vs OId.

**Comment by hansmosh** [comment]
@mattdodge A couple very minor comments on the block for you to review.

**Comment by mattdodge** [comment]
Added some unit tests, and did a little refactor. Also addressed review comments. Take a look @hansmosh and let me know if everything still looks good

**Pull request merged**